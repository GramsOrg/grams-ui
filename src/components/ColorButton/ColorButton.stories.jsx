import React from 'react';

import { action } from "@storybook/addon-actions";

import ColorButton from './ColorButton.tsx';

export default {
  title: 'General/ColorButton',
  component: ColorButton,
};

export const Default = () => <ColorButton color="blue" onClick={action("clicked")} />;

export const Red = () => <ColorButton color="red" onClick={action("clicked")} />;
