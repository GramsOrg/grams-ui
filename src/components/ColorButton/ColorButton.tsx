import React from 'react';

import './ColorButton.css';

export interface IColorButtonProps {
  color: string
  onClick?: (color: string) => void
}

const ColorButton = (props: IColorButtonProps) => {
  const {color, onClick} = props;
  return <button style={{color}} onClick={() => onClick && onClick(color)}>Color Button</button>
};

export default ColorButton;
