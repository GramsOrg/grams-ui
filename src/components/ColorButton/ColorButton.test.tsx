/**
 * @jest-environment jsdom
 */

import React from 'react';
import { describe, expect, test } from '@jest/globals';
import { render } from '@testing-library/react';

import ColorButton from './ColorButton';

describe('ColorButton Component', () => {
  test('it should match the snapshot', () => {
    const { asFragment } = render(<ColorButton color="blue" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
