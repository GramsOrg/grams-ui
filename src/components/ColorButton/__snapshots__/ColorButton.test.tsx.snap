// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorButton Component it should match the snapshot 1`] = `
<DocumentFragment>
  <button
    style="color: blue;"
  >
    Color Button
  </button>
</DocumentFragment>
`;
